src/App.tsx

```diff
@@ -1,19 +1,153 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.css';
-
+import todosFromServer from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedData: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: users.find((user) => user.id === todo.userId) || null,
+}));
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[];
+  title: string;
+  userId: number;
+  userSelected: boolean;
+  titleSelected: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todos: preparedData,
+    title: '',
+    userId: 0,
+    userSelected: true,
+    titleSelected: true,
+  };
+
+  addTodo = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+    const {
+      userId, title, userSelected, titleSelected,
+    } = this.state;
+
+    if (userSelected && titleSelected) {
+      this.setState(prevState => {
+        const createdTodo = {
+          user: users.find(user => user.id === userId) || null,
+          userId,
+          id: Math.max(...prevState.todos.map(item => item.id)) + 1,
+          title,
+          completed: false,
+        };
+
+        // eslint-disable-next-line no-console
+        console.log(createdTodo);
+
+        return {
+          todos: [...prevState.todos, createdTodo],
+          userId: 0,
+          title: '',
+        };
+      });
+    }
+  };
+
+  addTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      title: value,
+      titleSelected: true,
+    });
+  };
+
+  selectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      userId: +value,
+      userSelected: true,
+    });
+  };
+
+  checkFields = () => {
+    const { userId, title } = this.state;
+
+    if (title.length === 0) {
+      this.setState({
+        titleSelected: false,
+      });
+    }
+
+    if (userId === 0) {
+      this.setState({ userSelected: false });
+    }
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Add todo form</h1>
+
+        <form onSubmit={this.addTodo}>
+
+          <div className="mb-3">
+
+            <input
+              type="text"
+              className="form-control"
+              name="title"
+              onChange={this.addTitle}
+              value={this.state.title}
+              placeholder="Enter title"
+            />
+
+            {!this.state.titleSelected
+              && <p className="alert">Please enter the title!</p>}
```

Comment:
```suggestion
            {this.state.titleSelected
              || <p className="alert">Please enter the title!</p>}
```

and use destructuring